core_backend/app/urgency_rules/routers.py

```diff
@@ -0,0 +1,126 @@
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..auth.dependencies import get_current_fullaccess_user, get_current_readonly_user
+from ..auth.schemas import AuthenticatedUser
+from ..database import get_async_session
+from ..utils import setup_logger
+from .models import (
+    UrgencyRuleDB,
+    delete_urgency_rule_from_db,
+    get_urgency_rule_by_id_from_db,
+    get_urgency_rules_from_db,
+    save_urgency_rule_to_db,
+    update_urgency_rule_in_db,
+)
+from .schemas import UrgencyRuleCreate, UrgencyRuleRetrieve
+
+router = APIRouter(prefix="/urgency-rules", tags=["Urgency Rules"])
+logger = setup_logger(__name__)
+
+
+@router.post("/", response_model=UrgencyRuleRetrieve)
+async def create_urgency_rule(
+    urgency_rule: UrgencyRuleCreate,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> UrgencyRuleRetrieve:
+    """
+    Create a new urgency rule
+    """
+    urgency_rule_db = await save_urgency_rule_to_db(urgency_rule, asession)
+    return _convert_record_to_schema(urgency_rule_db)
+
+
+@router.get("/{urgency_rule_id}", response_model=UrgencyRuleRetrieve)
+async def get_urgency_rule(
+    urgency_rule_id: int,
+    full_access_user: Annotated[AuthenticatedUser, Depends(get_current_readonly_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> UrgencyRuleRetrieve:
+    """
+    Get a single urgency rule by id
+    """
+
+    urgency_rule_db = await get_urgency_rule_by_id_from_db(urgency_rule_id, asession)
+    if not urgency_rule_db:
+        raise HTTPException(
+            status_code=404, detail=f"Urgency Rule id `{urgency_rule_id}` not found"
+        )
+    return _convert_record_to_schema(urgency_rule_db)
+
+
+@router.delete("/{urgency_rule_id}")
+async def delete_urgency_rule(
+    urgency_rule_id: int,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> None:
+    """
+    Delete a single urgency rule by id
+    """
+
+    await delete_urgency_rule_from_db(urgency_rule_id, asession)
+
+
+@router.put("/{urgency_rule_id}", response_model=UrgencyRuleRetrieve)
+async def update_urgency_rule(
+    urgency_rule_id: int,
+    urgency_rule: UrgencyRuleCreate,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> UrgencyRuleRetrieve:
+    """
+    Update a single urgency rule by id
+    """
+    old_content = await get_urgency_rule_by_id_from_db(
+        urgency_rule_id,
+        asession,
+    )
+
+    if not old_content:
```

Comment:
```suggestion
    if not old_urgency_rule:
```suggestion
    if not old_content:
```
```